coap-service/coap_service_api.h

```diff
@@ -47,6 +47,8 @@ extern "C" {
 #define COAP_SERVICE_OPTIONS_EPHEMERAL_PORT     0x04
 /** Coap interface selected as socket interface */
 #define COAP_SERVICE_OPTIONS_SELECT_SOCKET_IF   0x08
+/** Register to COAP multicast groups */
```

Comment:
I'm conflicted on this flag. I think that joining the groups should be default behaviour (RFC 7390 - "All CoAP server nodes SHOULD join the "All CoAP Nodes" multicast group"), in which case the flag polarity would be reversed.

On the other hand "negative" flags can be confusing.